acvm/src/lib.rs

```diff
@@ -101,6 +101,15 @@ pub trait PartialWitnessGenerator {
         input: &FunctionInput,
         outputs: &[Witness],
     ) -> Result<OpcodeResolution, OpcodeResolutionError>;
+    fn verify_proof(
```

Comment:
I forget the conclusion of our discussion re having this done inside of the backend. 

I'll information dump for context.

Removing it gives an imbalance because the RecursiveAggregation opcode will be the only one without a solver method.

Keeping it means that the solver will either directly or indirectly need a backend (unless the solver implements it to return true) + it will be a component that will now need to be setted when we attempt to solve.

The root root problem being that there is no standard way to do recursive aggregation, so its not possible to implement it here like the other opcodes. However, since these witnesses are never used by the business logic in the program (fed into another opcode) we can optimistically assume that it verifies true and ignore it.

The conversation over slack was that we needed to set the output values `output_aggregation_object`using this method, but it was noted that we do not need to set these until we get to the backend since they are never used in other places in the IR, so we could remove this from the ACVM layer altogether.

Reply:
I think we are safe to remove this from the ACVM layer. However, it does feel weird for `RecursiveAggregation` to not have a solver method. I can remove this from the ACVM for now and leave a comment describing why it is unnecessary and our choice to always have the recursion opcode pass execution during simulation.